### PR TITLE
48 simplify notch approximation lawbinned

### PR DESCRIPTION
**Pull request by maierbn**

### CODE REVIEW

File: src/pylife/materiallaws/notch_approximation_law.py [hunk 1]
```diff
@@ -674,140 +581,458 @@ def stress_secondary_branch(self, delta_load, *, rtol=1e-5, tol=1e-6):
...
-            index = self._lut_secondary_branch.delta_load.searchsorted(np.abs(delta_load.values))-1   # "-1", transform to zero-based indices
+        load = load.fillna(0)
+        sign = sign.fillna(0)
+        
+        index = self._lut_primary_branch.load.searchsorted(np.abs(load.values))-1   # "-1", transform to zero-based indices
+
+        # raise error if requested load is higher than initialized maximum absolute load
+        if np.any(index+1 >= len(self._lut_primary_branch)):
+            raise ValueError(f"Binned class is initialized with a maximum absolute load of {self._maximum_absolute_load}, "\
+                                f" but a higher absolute load value of |{load.max()}| is requested (in stress()).")
```
Comment (johannes-mueller) [inline src/pylife/materiallaws/notch_approximation_law.py:701]: Maybe we can `._find_load_index_or_raise(load)` which finds the `index` of the load or raises if any of them is outside. Would remove code duplication and help readability.

Or at least introduce a method `._raise_if_requested_load_higher_than_max_abs_load(index, max_abs_load, max_load)`

**Review by johannes-mueller — CHANGES_REQUESTED** [review]
Quite an improvement, thanks. Please take a look at the one comment in the code.

**Review by johannes-mueller — APPROVED** [review]
Squash?

**Comment by maierbn** [comment]
I forgot the Sign-off again. Does it gets fixed with a squash? Then yes, please :-)

**Comment by johannes-mueller** [comment]
Ok, thanks.

**Pull request merged**